src/NuGet.Clients/NuGet.VisualStudio/Extensibility/IVsFrameworkCompatibility2.cs

```diff
@@ -30,6 +30,7 @@ public interface IVsFrameworkCompatibility2 : IVsFrameworkCompatibility
         /// <param name="frameworks">The list of frameworks to choose from.</param>
         /// <exception cref="ArgumentException">If any of the arguments are <c>null</c>.</exception>
         /// <returns>The nearest framework.</returns>
+        [Obsolete("This API does not support .NET 5 and higher target frameworks with platforms. Use IVsFrameworkCompatibility3 instead.")]
```

Comment:
Why not deprecate the whole `IVsFrameworkCompatibility2` interface? All methods (1) of this interface will be deprecated in this PR.